### ISSUE TITLE
Include solo players in Faction Grid Count limits

### ISSUE THREAD

**Issue by zrisher** [stackcollision/GardenConquest#15]
If we're allowing a certain number of basic unlicensed grids, I think it makes most sense to distribute it per player instead of per faction. This would remove any disadvantage from being in a faction and would enable us to apply class count limits to unfactioned (solo) players more easily I.e.:

```
Limit of 2 unlicensed grids per player:
Faction of 5 people - 10 unlicensed grids
Faction of 2 people - 4 unlicensed grids
Faction of 1 person - 2 unlicensed grids
Solo player without faction - 2 unlicensed grids
```

This would solve the problem in `GridEnforcer.checkClassAllowed` where we can't get the grid limit for a solo player because we don't keep a FactionFleet for them - we should keep one of these for every solo player, just treat them as a faction of 1.

This would be nice to have in the beta to minimize confusion later since the limit changes would affect everyone, both those in factions and not.

**Comment by stackcollision** [comment]
This is ripe for abuse.  10 unlicensed ships for a 5 person faction is an entire fleet.

**Comment by zrisher** [comment]
My understanding was that unlicensed ships were small, simple, and basically inferior in every way to a licensed grid wrt gameplay. A faction wouldn't bother having 10 unlicensed ships because they'd way rather have 2 licensed ones. But until they have some license blocks, it's nice they're not constrained to less ships then they have people simply because they joined a faction with more than 1 other person.

**Comment by zrisher** [comment]
How about 1 unlicensed small ship per player, regardless of faction, but only 1 unlicensed platform/large ship per faction?

**Comment by zrisher** [comment]
For the beta, we'll need to at least have some sort of limit for people not in factions. Currently, they can build as many as they want:

```
// Blocks/GridEnforcer.cs

private bool checkClassAllowed(HullClass.CLASS c) {
            // QUESTION: maybe players not in factions shouldn't be able to classify?
            if (m_OwningFaction == null)
                return true;
```

**Comment by zrisher** [comment]
Closed in favor of #26
